### PR TITLE
Windows branch merge

### PR DESCRIPTION
**Pull request by kuba160**
Changes:
- Update premake build files 
    - use `shared/eqpreset.c` to build gtk plugins
    - fix build order
    - upstream premake does (did) not accept piped commands; now a separate script `scripts/windows_postbuild.sh` is used
    - fix for removed .svg file in `pixmaps/`
    - alac plugin added
- In `playlist.c` `d_type` block section is dependent on the system. As [GNU libc documentation](https://www.gnu.org/software/libc/manual/html_node/Directory-Entries.html) states `_DIRENT_HAVE_D_TYPE` would be the proper way to check for D_TYPE extension presence (pinging @EubiquitouS #2318)
- reorganization of `shared/windows/utils.c` and fix for msys2 gtk changes
- `gtk_file_chooser_set_current_folder_uri` disabled for windows build with native file chooser. It is possible that this fix should be applied on linux version as well (with native file chooser), although I didn't test it yet

### CODE REVIEW

**Comment by kuba160** [comment]
Also one question: is `shared/eqpreset.c` released under ZLIB license? Current header does not indicate that.

**Comment by EubiquitouS** [comment]
> As GNU libc documentation states `_DIRENT_HAVE_D_TYPE` would be the proper way to check for D_TYPE extension presence 

This looks glibc-specific though. On my OpenBSD box, `_DIRENT_HAVE_D_TYPE` is nowhere to be seen, despite `dirent` having `d_type` as a member (it is a BSD extension, after all).

**Comment by Alexey-Yakovenko** [comment]
> As GNU libc documentation states _DIRENT_HAVE_D_TYPE would be the proper way to check for D_TYPE extension presence

Sounds like the most reliable way would be to test for it via compiling a test program using autotools.
But let's keep it the way it was for now. We are using 3 different build systems at the moment, and not all of them can do such test.

**Comment by Alexey-Yakovenko** [comment]
> Also one question: is shared/eqpreset.c released under ZLIB license? Current header does not indicate that.

Fixed. It is zlib.

**Comment by kuba160** [comment]
This source states that `#if defined(DTTOIF)` would be universal way to detect d_type: 
> While d_type is a non-standard extension to struct dirent, only glibc signals its presence with another macro _DIRENT_HAVE_D_TYPE. However, any platform with d_type also includes a way to convert to mode_t values using the macro DTTOIF, so we can check for that alone and still be confident that the d_type member exists.

https://reviews.llvm.org/D64940

File: plugins/gtkui/fileman.c [hunk 1]
```diff
@@ -642,9 +642,12 @@ show_file_chooser (const gchar          *title,
     }
 
     // restore folder
+    // Windows: restore done by gtk? (todo check linux)
+    #if defined USE_GTK_NATIVE_FILE_CHOOSER && defined __MINGW32__
```
[the remark this answers is not in the document]
Comment (kuba160) [reply plugins/gtkui/fileman.c]: Looks like there was some regression in setting current folder in gtk, looks fine now

File: scripts/windows_postbuild.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# Translations
+for i in po/*.gmo ; do
+	base=`basename $i .gmo`
+	mkdir -pv $1/locale/$base/LC_MESSAGES
+	cp -uv $i $1/locale/$base/LC_MESSAGES/deadbeef.mo
+done
+cp -uv translation/help.ru.txt  $1/doc/
+
+# Libraries
+rm -fv $1/plugins/*.lib | true
```
[the remark this answers is not in the document]
Comment (kuba160) [reply scripts/windows_postbuild.sh]: I put double quotes surrounding `$1`, is it enough? I am not that good with shell scripts.

[the remark this answers is not in the document]
Comment (Alexey-Yakovenko) [reply scripts/windows_postbuild.sh]: Yes, this is enough, thanks!

**Pull request merged**